<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Seurat | Portfolio FFR</title>
  <meta name="description" content="3 Seurat | Portfolio FFR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Seurat | Portfolio FFR" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Seurat | Portfolio FFR" />
  
  
  

<meta name="author" content="FFR" />


<meta name="date" content="2024-11-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="curriculum-vitae-ffr.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href=""></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="crohn_dissease.html"><a href="crohn_dissease.html"><i class="fa fa-check"></i><b>1</b> Crohn_dissease</a></li>
<li class="chapter" data-level="2" data-path="curriculum-vitae-ffr.html"><a href="curriculum-vitae-ffr.html"><i class="fa fa-check"></i><b>2</b> Curriculum vitae FFR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="curriculum-vitae-ffr.html"><a href="curriculum-vitae-ffr.html#personalia-verander-zodat-er-meer-is-over-bioinfo"><i class="fa fa-check"></i><b>2.1</b> Personalia verander zodat er meer is over bioinfo</a></li>
<li class="chapter" data-level="2.2" data-path="curriculum-vitae-ffr.html"><a href="curriculum-vitae-ffr.html#opleiding"><i class="fa fa-check"></i><b>2.2</b> Opleiding</a></li>
<li class="chapter" data-level="2.3" data-path="curriculum-vitae-ffr.html"><a href="curriculum-vitae-ffr.html#werkervaring"><i class="fa fa-check"></i><b>2.3</b> Werkervaring</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="seurat.html"><a href="seurat.html"><i class="fa fa-check"></i><b>3</b> Seurat</a></li>
<li class="divider"></li>
<li><a href="" target="blank"></a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Portfolio FFR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="seurat" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Seurat<a href="seurat.html#seurat" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The Seurat package is an R package that is designed for analysis and visualization of single-cell RNA sequencing (scRNA-seq) data. It is possible to do data normalization, dimensionality reduction (like, PCA and UMAP) and for other forms of analyses (omic if I get to it).</p>
<p>I will be using this (<a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial" class="uri">https://satijalab.org/seurat/articles/pbmc3k_tutorial</a>) tutorial to start learning the Seurat package.</p>
<p>I quote “we will be analyzing the a dataset of Peripheral Blood Mononuclear Cells (PBMC) freely available from 10X Genomics. There are 2,700 single cells that were sequenced on the Illumina NextSeq 500. The raw data can be found here(<a href="https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz" class="uri">https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz</a>).”</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="seurat.html#cb63-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb63-2"><a href="seurat.html#cb63-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb63-3"><a href="seurat.html#cb63-3" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb63-4"><a href="seurat.html#cb63-4" tabindex="-1"></a><span class="co">#install.packages(&quot;Seurat&quot;)</span></span>
<span id="cb63-5"><a href="seurat.html#cb63-5" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb63-6"><a href="seurat.html#cb63-6" tabindex="-1"></a><span class="co">#install.packages(&quot;patchwork&quot;)</span></span>
<span id="cb63-7"><a href="seurat.html#cb63-7" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb63-8"><a href="seurat.html#cb63-8" tabindex="-1"></a></span>
<span id="cb63-9"><a href="seurat.html#cb63-9" tabindex="-1"></a><span class="co"># Load the PBMC dataset</span></span>
<span id="cb63-10"><a href="seurat.html#cb63-10" tabindex="-1"></a>pbmc.data <span class="ot">&lt;-</span> <span class="fu">Read10X</span>(<span class="at">data.dir =</span> <span class="st">&quot;Rmd_files/Seurat_10x_data/filtered_gene_bc_matrices/hg19/&quot;</span>)</span>
<span id="cb63-11"><a href="seurat.html#cb63-11" tabindex="-1"></a></span>
<span id="cb63-12"><a href="seurat.html#cb63-12" tabindex="-1"></a><span class="co"># Initialize the Seurat object with the raw (non-normalized data).</span></span>
<span id="cb63-13"><a href="seurat.html#cb63-13" tabindex="-1"></a>pbmc <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> pbmc.data, <span class="at">project =</span> <span class="st">&quot;pbmc3k&quot;</span>, <span class="at">min.cells =</span> <span class="dv">3</span>, <span class="at">min.features =</span> <span class="dv">200</span>)</span>
<span id="cb63-14"><a href="seurat.html#cb63-14" tabindex="-1"></a>pbmc</span></code></pre></div>
<pre><code>## An object of class Seurat 
## 13714 features across 2700 samples within 1 assay 
## Active assay: RNA (13714 features, 0 variable features)
##  1 layer present: counts</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="seurat.html#cb65-1" tabindex="-1"></a><span class="co"># Lets examine a few genes in the first thirty cells</span></span>
<span id="cb65-2"><a href="seurat.html#cb65-2" tabindex="-1"></a>pbmc.data[<span class="fu">c</span>(<span class="st">&quot;CD3D&quot;</span>, <span class="st">&quot;TCL1A&quot;</span>, <span class="st">&quot;MS4A1&quot;</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>]</span></code></pre></div>
<pre><code>## 3 x 30 sparse Matrix of class &quot;dgCMatrix&quot;</code></pre>
<pre><code>##   [[ suppressing 30 column names &#39;AAACATACAACCAC-1&#39;, &#39;AAACATTGAGCTAC-1&#39;, &#39;AAACATTGATCAGC-1&#39; ... ]]</code></pre>
<pre><code>##                                                                    
## CD3D  4 . 10 . . 1 2 3 1 . . 2 7 1 . . 1 3 . 2  3 . . . . . 3 4 1 5
## TCL1A . .  . . . . . . 1 . . . . . . . . . . .  . 1 . . . . . . . .
## MS4A1 . 6  . . . . . . 1 1 1 . . . . . . . . . 36 1 2 . . 2 . . . .</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="seurat.html#cb69-1" tabindex="-1"></a>dense.size <span class="ot">&lt;-</span> <span class="fu">object.size</span>(<span class="fu">as.matrix</span>(pbmc.data))</span>
<span id="cb69-2"><a href="seurat.html#cb69-2" tabindex="-1"></a><span class="co">#object.size(): Calculates the memory size of the dense matrix in bytes.</span></span>
<span id="cb69-3"><a href="seurat.html#cb69-3" tabindex="-1"></a><span class="co">#as.matrix(pbmc.data): Converts pbmc.data to a dense matrix format were as normally a parse matrix is used because it takes up less storage/memory </span></span>
<span id="cb69-4"><a href="seurat.html#cb69-4" tabindex="-1"></a>sparse.size <span class="ot">&lt;-</span> <span class="fu">object.size</span>(pbmc.data)</span>
<span id="cb69-5"><a href="seurat.html#cb69-5" tabindex="-1"></a>sparse.size</span></code></pre></div>
<pre><code>## 29905192 bytes</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="seurat.html#cb71-1" tabindex="-1"></a><span class="co">#print parse matrix</span></span>
<span id="cb71-2"><a href="seurat.html#cb71-2" tabindex="-1"></a>dense.size</span></code></pre></div>
<pre><code>## 709591472 bytes</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="seurat.html#cb73-1" tabindex="-1"></a><span class="co">#print dense matrix</span></span>
<span id="cb73-2"><a href="seurat.html#cb73-2" tabindex="-1"></a>dense.size<span class="sc">/</span>sparse.size</span></code></pre></div>
<pre><code>## 23.7 bytes</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="seurat.html#cb75-1" tabindex="-1"></a><span class="co">#The dense matrix is 23.7 times bigger than the parse</span></span></code></pre></div>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="seurat.html#cb76-1" tabindex="-1"></a><span class="co"># The [[ operator can add columns to object metadata. This is a great place to stash QC stats</span></span>
<span id="cb76-2"><a href="seurat.html#cb76-2" tabindex="-1"></a>pbmc[[<span class="st">&quot;percent.mt&quot;</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(pbmc, <span class="at">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</span>
<span id="cb76-3"><a href="seurat.html#cb76-3" tabindex="-1"></a></span>
<span id="cb76-4"><a href="seurat.html#cb76-4" tabindex="-1"></a><span class="co"># Show QC metrics for the first 5 cells</span></span>
<span id="cb76-5"><a href="seurat.html#cb76-5" tabindex="-1"></a><span class="fu">head</span>(pbmc<span class="sc">@</span>meta.data, <span class="dv">5</span>)</span></code></pre></div>
<pre><code>##                  orig.ident nCount_RNA nFeature_RNA percent.mt
## AAACATACAACCAC-1     pbmc3k       2419          779  3.0177759
## AAACATTGAGCTAC-1     pbmc3k       4903         1352  3.7935958
## AAACATTGATCAGC-1     pbmc3k       3147         1129  0.8897363
## AAACCGTGCTTCCG-1     pbmc3k       2639          960  1.7430845
## AAACCGTGTATGCG-1     pbmc3k        980          521  1.2244898</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="seurat.html#cb78-1" tabindex="-1"></a><span class="co"># Visualize QC metrics as a violin plot</span></span>
<span id="cb78-2"><a href="seurat.html#cb78-2" tabindex="-1"></a><span class="fu">VlnPlot</span>(pbmc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## Warning: Default search for &quot;data&quot; layer in &quot;RNA&quot; assay yielded no results;
## utilizing &quot;counts&quot; layer instead.</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="seurat.html#cb80-1" tabindex="-1"></a><span class="co"># FeatureScatter is typically used to visualize feature-feature relationships, but can be used</span></span>
<span id="cb80-2"><a href="seurat.html#cb80-2" tabindex="-1"></a><span class="co"># for anything calculated by the object, i.e. columns in object metadata, PC scores etc.</span></span>
<span id="cb80-3"><a href="seurat.html#cb80-3" tabindex="-1"></a></span>
<span id="cb80-4"><a href="seurat.html#cb80-4" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(pbmc, <span class="at">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;percent.mt&quot;</span>, <span class="at">plot.cor =</span> <span class="cn">TRUE</span>)</span>
<span id="cb80-5"><a href="seurat.html#cb80-5" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(pbmc, <span class="at">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>, <span class="at">plot.cor =</span> <span class="cn">TRUE</span>)</span>
<span id="cb80-6"><a href="seurat.html#cb80-6" tabindex="-1"></a>plot1 <span class="sc">+</span> plot2</span></code></pre></div>
<p><img src="_main_files/figure-html/scatterplots-1.png" width="672" /></p>
<p>There are various ways of generating qualtity control metrics. The tutorial suggests filtering away feature counts over 2500 and under 200 and mitochondrial counts higher than 5%.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="seurat.html#cb81-1" tabindex="-1"></a><span class="co">#selecting the subset of data wanted</span></span>
<span id="cb81-2"><a href="seurat.html#cb81-2" tabindex="-1"></a>subset_pbmc <span class="ot">&lt;-</span> <span class="fu">subset</span>(pbmc, <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">200</span> <span class="sc">&amp;</span> nFeature_RNA <span class="sc">&lt;</span> <span class="dv">2500</span> <span class="sc">&amp;</span> percent.mt <span class="sc">&lt;</span> <span class="dv">5</span>)</span>
<span id="cb81-3"><a href="seurat.html#cb81-3" tabindex="-1"></a></span>
<span id="cb81-4"><a href="seurat.html#cb81-4" tabindex="-1"></a><span class="co">#normalizing the data as to make the numbers more human readable</span></span>
<span id="cb81-5"><a href="seurat.html#cb81-5" tabindex="-1"></a>norm_subset_pbmc <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(subset_pbmc)</span></code></pre></div>
<pre><code>## Normalizing layer: counts</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="seurat.html#cb83-1" tabindex="-1"></a>FVF_pbmc <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(norm_subset_pbmc, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>)</span></code></pre></div>
<pre><code>## Finding variable features for layer counts</code></pre>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="seurat.html#cb85-1" tabindex="-1"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb85-2"><a href="seurat.html#cb85-2" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">VariableFeatures</span>(FVF_pbmc))</span>
<span id="cb85-3"><a href="seurat.html#cb85-3" tabindex="-1"></a></span>
<span id="cb85-4"><a href="seurat.html#cb85-4" tabindex="-1"></a><span class="co"># plot variable features with and without labels</span></span>
<span id="cb85-5"><a href="seurat.html#cb85-5" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> <span class="fu">VariableFeaturePlot</span>(FVF_pbmc)</span>
<span id="cb85-6"><a href="seurat.html#cb85-6" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> <span class="fu">LabelPoints</span>(<span class="at">plot =</span> plot3, <span class="at">points =</span> top10, <span class="at">repel =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## When using repel, set xnudge and ynudge to 0 for optimal results</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="seurat.html#cb87-1" tabindex="-1"></a>plot3 <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span> plot4 </span></code></pre></div>
<pre><code>## Warning in scale_x_log10(): log-10 transformation introduced
## infinite values.
## log-10 transformation introduced
## infinite values.</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="seurat.html#cb89-1" tabindex="-1"></a><span class="co">#retrieve all gene names</span></span>
<span id="cb89-2"><a href="seurat.html#cb89-2" tabindex="-1"></a>all.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(FVF_pbmc)</span>
<span id="cb89-3"><a href="seurat.html#cb89-3" tabindex="-1"></a></span>
<span id="cb89-4"><a href="seurat.html#cb89-4" tabindex="-1"></a><span class="co">#Scales and centers features in the dataset.</span></span>
<span id="cb89-5"><a href="seurat.html#cb89-5" tabindex="-1"></a>scaled_FVF_pbmc <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(FVF_pbmc, <span class="at">features =</span> all.genes)</span></code></pre></div>
<pre><code>## Centering and scaling data matrix</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="seurat.html#cb91-1" tabindex="-1"></a><span class="co">#Run a PCA dimensionality reduction. For details about stored PCA calculation parameters</span></span>
<span id="cb91-2"><a href="seurat.html#cb91-2" tabindex="-1"></a>PCA_pbmc <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(scaled_FVF_pbmc, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> scaled_FVF_pbmc))</span></code></pre></div>
<pre><code>## PC_ 1 
## Positive:  CST3, TYROBP, LST1, AIF1, FTL, FTH1, LYZ, FCN1, S100A9, TYMP 
##     FCER1G, CFD, LGALS1, S100A8, CTSS, LGALS2, SERPINA1, IFITM3, SPI1, CFP 
##     PSAP, IFI30, SAT1, COTL1, S100A11, NPC2, GRN, LGALS3, GSTP1, PYCARD 
## Negative:  MALAT1, LTB, IL32, IL7R, CD2, B2M, ACAP1, CD27, STK17A, CTSW 
##     CD247, GIMAP5, AQP3, CCL5, SELL, TRAF3IP3, GZMA, MAL, CST7, ITM2A 
##     MYC, GIMAP7, HOPX, BEX2, LDLRAP1, GZMK, ETS1, ZAP70, TNFAIP8, RIC3 
## PC_ 2 
## Positive:  CD79A, MS4A1, TCL1A, HLA-DQA1, HLA-DQB1, HLA-DRA, LINC00926, CD79B, HLA-DRB1, CD74 
##     HLA-DMA, HLA-DPB1, HLA-DQA2, CD37, HLA-DRB5, HLA-DMB, HLA-DPA1, FCRLA, HVCN1, LTB 
##     BLNK, P2RX5, IGLL5, IRF8, SWAP70, ARHGAP24, FCGR2B, SMIM14, PPP1R14A, C16orf74 
## Negative:  NKG7, PRF1, CST7, GZMB, GZMA, FGFBP2, CTSW, GNLY, B2M, SPON2 
##     CCL4, GZMH, FCGR3A, CCL5, CD247, XCL2, CLIC3, AKR1C3, SRGN, HOPX 
##     TTC38, APMAP, CTSC, S100A4, IGFBP7, ANXA1, ID2, IL32, XCL1, RHOC 
## PC_ 3 
## Positive:  HLA-DQA1, CD79A, CD79B, HLA-DQB1, HLA-DPB1, HLA-DPA1, CD74, MS4A1, HLA-DRB1, HLA-DRA 
##     HLA-DRB5, HLA-DQA2, TCL1A, LINC00926, HLA-DMB, HLA-DMA, CD37, HVCN1, FCRLA, IRF8 
##     PLAC8, BLNK, MALAT1, SMIM14, PLD4, P2RX5, IGLL5, LAT2, SWAP70, FCGR2B 
## Negative:  PPBP, PF4, SDPR, SPARC, GNG11, NRGN, GP9, RGS18, TUBB1, CLU 
##     HIST1H2AC, AP001189.4, ITGA2B, CD9, TMEM40, PTCRA, CA2, ACRBP, MMD, TREML1 
##     NGFRAP1, F13A1, SEPT5, RUFY1, TSC22D1, MPP1, CMTM5, RP11-367G6.3, MYL9, GP1BA 
## PC_ 4 
## Positive:  HLA-DQA1, CD79B, CD79A, MS4A1, HLA-DQB1, CD74, HIST1H2AC, HLA-DPB1, PF4, SDPR 
##     TCL1A, HLA-DRB1, HLA-DPA1, HLA-DQA2, PPBP, HLA-DRA, LINC00926, GNG11, SPARC, HLA-DRB5 
##     GP9, AP001189.4, CA2, PTCRA, CD9, NRGN, RGS18, CLU, TUBB1, GZMB 
## Negative:  VIM, IL7R, S100A6, IL32, S100A8, S100A4, GIMAP7, S100A10, S100A9, MAL 
##     AQP3, CD2, CD14, FYB, LGALS2, GIMAP4, ANXA1, CD27, FCN1, RBP7 
##     LYZ, S100A11, GIMAP5, MS4A6A, S100A12, FOLR3, TRABD2A, AIF1, IL8, IFI6 
## PC_ 5 
## Positive:  GZMB, NKG7, S100A8, FGFBP2, GNLY, CCL4, CST7, PRF1, GZMA, SPON2 
##     GZMH, S100A9, LGALS2, CCL3, CTSW, XCL2, CD14, CLIC3, S100A12, RBP7 
##     CCL5, MS4A6A, GSTP1, FOLR3, IGFBP7, TYROBP, TTC38, AKR1C3, XCL1, HOPX 
## Negative:  LTB, IL7R, CKB, VIM, MS4A7, AQP3, CYTIP, RP11-290F20.3, SIGLEC10, HMOX1 
##     LILRB2, PTGES3, MAL, CD27, HN1, CD2, GDI2, CORO1B, ANXA5, TUBA1B 
##     FAM110A, ATP1A1, TRADD, PPA1, CCDC109B, ABRACL, CTD-2006K23.1, WARS, VMO1, FYB</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="seurat.html#cb93-1" tabindex="-1"></a><span class="co"># Examine and visualize PCA results a few different ways</span></span>
<span id="cb93-2"><a href="seurat.html#cb93-2" tabindex="-1"></a><span class="fu">print</span>(PCA_pbmc[[<span class="st">&quot;pca&quot;</span>]], <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">nfeatures =</span> <span class="dv">5</span>)</span></code></pre></div>
<pre><code>## PC_ 1 
## Positive:  CST3, TYROBP, LST1, AIF1, FTL 
## Negative:  MALAT1, LTB, IL32, IL7R, CD2 
## PC_ 2 
## Positive:  CD79A, MS4A1, TCL1A, HLA-DQA1, HLA-DQB1 
## Negative:  NKG7, PRF1, CST7, GZMB, GZMA 
## PC_ 3 
## Positive:  HLA-DQA1, CD79A, CD79B, HLA-DQB1, HLA-DPB1 
## Negative:  PPBP, PF4, SDPR, SPARC, GNG11 
## PC_ 4 
## Positive:  HLA-DQA1, CD79B, CD79A, MS4A1, HLA-DQB1 
## Negative:  VIM, IL7R, S100A6, IL32, S100A8 
## PC_ 5 
## Positive:  GZMB, NKG7, S100A8, FGFBP2, GNLY 
## Negative:  LTB, IL7R, CKB, VIM, MS4A7</code></pre>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="seurat.html#cb95-1" tabindex="-1"></a><span class="co">#Visualize the top genes and the first two principle components </span></span>
<span id="cb95-2"><a href="seurat.html#cb95-2" tabindex="-1"></a><span class="fu">VizDimLoadings</span>(PCA_pbmc, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/PCA-1.png" width="672" /></p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="seurat.html#cb96-1" tabindex="-1"></a><span class="co">#A PCA plot depicting the each cell and where they get position measured against PC1 and PC2</span></span>
<span id="cb96-2"><a href="seurat.html#cb96-2" tabindex="-1"></a><span class="fu">DimPlot</span>(PCA_pbmc, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="seurat.html#cb97-1" tabindex="-1"></a><span class="fu">DimHeatmap</span>(PCA_pbmc, <span class="at">dims =</span> <span class="dv">1</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="seurat.html#cb98-1" tabindex="-1"></a><span class="co">#By suggesting to the DimHeatmap function to look at PC1 through 15 it will try to fit 15 heatmaps based on the PC previously calculated in PCA_pbmc</span></span>
<span id="cb98-2"><a href="seurat.html#cb98-2" tabindex="-1"></a><span class="fu">DimHeatmap</span>(PCA_pbmc, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-20-1.png" width="672" />
## Dementionality</p>
<p>Determining the amount of PC that need to be used is important as to not have a overly computationally expensive. The more PC’s used has diminishing marginal results. This can be done visually by using ElbowPlot(). Alot of the deviation in the graphs can be explained in the first 10 of the principle coordinates. The differences produced in PC 10 - 20 are less distinctionable from background than the first 10.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="seurat.html#cb99-1" tabindex="-1"></a><span class="fu">ElbowPlot</span>(PCA_pbmc)</span></code></pre></div>
<p><img src="_main_files/figure-html/dementionality%20visualization-1.png" width="672" /></p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="seurat.html#cb100-1" tabindex="-1"></a>findneighbour_pbmc <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(PCA_pbmc, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>) <span class="co"># selecting the top 10 pc&#39;s</span></span></code></pre></div>
<pre><code>## Computing nearest neighbor graph</code></pre>
<pre><code>## Computing SNN</code></pre>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="seurat.html#cb103-1" tabindex="-1"></a>clustered_pbmc <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(findneighbour_pbmc, <span class="at">resolution =</span> <span class="fl">0.5</span>) <span class="co"># clustering the PC&#39;s</span></span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 2638
## Number of edges: 95927
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.8728
## Number of communities: 9
## Elapsed time: 0 seconds</code></pre>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="seurat.html#cb105-1" tabindex="-1"></a>rUMAP_pbmc <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(clustered_pbmc, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span></code></pre></div>
<pre><code>## 11:32:16 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
<pre><code>## 11:32:16 Read 2638 rows and found 10 numeric columns</code></pre>
<pre><code>## 11:32:16 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
<pre><code>## 11:32:16 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>## [----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>## **************************************************|
## 11:32:17 Writing NN index file to temp file C:\Users\ffrij\AppData\Local\Temp\Rtmp4MRHuY\file3d10602c126f
## 11:32:17 Searching Annoy index using 1 thread, search_k = 3000
## 11:32:17 Annoy recall = 100%
## 11:32:18 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
## 11:32:19 Initializing from normalized Laplacian + noise (using RSpectra)
## 11:32:19 Commencing optimization for 500 epochs, with 105140 positive edges
## 11:32:25 Optimization finished</code></pre>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="seurat.html#cb113-1" tabindex="-1"></a><span class="co"># note that you can set `label = TRUE` or use the LabelClusters function to help label</span></span>
<span id="cb113-2"><a href="seurat.html#cb113-2" tabindex="-1"></a><span class="co"># individual clusters</span></span>
<span id="cb113-3"><a href="seurat.html#cb113-3" tabindex="-1"></a><span class="fu">DimPlot</span>(rUMAP_pbmc, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="seurat.html#cb114-1" tabindex="-1"></a>cluster2.markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(rUMAP_pbmc, <span class="at">ident.1 =</span> <span class="dv">2</span>)</span>
<span id="cb114-2"><a href="seurat.html#cb114-2" tabindex="-1"></a><span class="fu">head</span>(cluster2.markers, <span class="at">n =</span> <span class="dv">5</span>)</span></code></pre></div>
<pre><code>##             p_val avg_log2FC pct.1 pct.2    p_val_adj
## IL32 2.892340e-90  1.3070772 0.947 0.465 3.966555e-86
## LTB  1.060121e-86  1.3312674 0.981 0.643 1.453850e-82
## CD3D 8.794641e-71  1.0597620 0.922 0.432 1.206097e-66
## IL7R 3.516098e-68  1.4377848 0.750 0.326 4.821977e-64
## LDHB 1.642480e-67  0.9911924 0.954 0.614 2.252497e-63</code></pre>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="seurat.html#cb116-1" tabindex="-1"></a><span class="co"># find all markers distinguishing cluster 5 from clusters 0 and 3</span></span>
<span id="cb116-2"><a href="seurat.html#cb116-2" tabindex="-1"></a>cluster5.markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(rUMAP_pbmc, <span class="at">ident.1 =</span> <span class="dv">5</span>, <span class="at">ident.2 =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">3</span>))</span>
<span id="cb116-3"><a href="seurat.html#cb116-3" tabindex="-1"></a><span class="fu">head</span>(cluster5.markers, <span class="at">n =</span> <span class="dv">5</span>)</span></code></pre></div>
<pre><code>##                       p_val avg_log2FC pct.1 pct.2     p_val_adj
## FCGR3A        8.246578e-205   6.794969 0.975 0.040 1.130936e-200
## IFITM3        1.677613e-195   6.192558 0.975 0.049 2.300678e-191
## CFD           2.401156e-193   6.015172 0.938 0.038 3.292945e-189
## CD68          2.900384e-191   5.530330 0.926 0.035 3.977587e-187
## RP11-290F20.3 2.513244e-186   6.297999 0.840 0.017 3.446663e-182</code></pre>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="seurat.html#cb118-1" tabindex="-1"></a>pbmc.markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(rUMAP_pbmc, <span class="at">only.pos =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Calculating cluster 0</code></pre>
<pre><code>## Calculating cluster 1</code></pre>
<pre><code>## Calculating cluster 2</code></pre>
<pre><code>## Calculating cluster 3</code></pre>
<pre><code>## Calculating cluster 4</code></pre>
<pre><code>## Calculating cluster 5</code></pre>
<pre><code>## Calculating cluster 6</code></pre>
<pre><code>## Calculating cluster 7</code></pre>
<pre><code>## Calculating cluster 8</code></pre>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="seurat.html#cb128-1" tabindex="-1"></a>pbmc.markers <span class="sc">%&gt;%</span></span>
<span id="cb128-2"><a href="seurat.html#cb128-2" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb128-3"><a href="seurat.html#cb128-3" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## # A tibble: 7,019 × 7
## # Groups:   cluster [9]
##        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene     
##        &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;    
##  1 3.75e-112       1.21 0.912 0.592 5.14e-108 0       LDHB     
##  2 9.57e- 88       2.40 0.447 0.108 1.31e- 83 0       CCR7     
##  3 1.15e- 76       1.06 0.845 0.406 1.58e- 72 0       CD3D     
##  4 1.12e- 54       1.04 0.731 0.4   1.54e- 50 0       CD3E     
##  5 1.35e- 51       2.14 0.342 0.103 1.86e- 47 0       LEF1     
##  6 1.94e- 47       1.20 0.629 0.359 2.66e- 43 0       NOSIP    
##  7 2.81e- 44       1.53 0.443 0.185 3.85e- 40 0       PIK3IP1  
##  8 6.27e- 43       1.99 0.33  0.112 8.60e- 39 0       PRKCQ-AS1
##  9 1.16e- 40       2.70 0.2   0.04  1.59e- 36 0       FHIT     
## 10 1.34e- 34       1.96 0.268 0.087 1.84e- 30 0       MAL      
## # ℹ 7,009 more rows</code></pre>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="seurat.html#cb130-1" tabindex="-1"></a><span class="co"># find markers for every cluster compared to all remaining cells, report only the positive ones</span></span>
<span id="cb130-2"><a href="seurat.html#cb130-2" tabindex="-1"></a></span>
<span id="cb130-3"><a href="seurat.html#cb130-3" tabindex="-1"></a>pbmc.markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(rUMAP_pbmc, <span class="at">only.pos =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Calculating cluster 0</code></pre>
<pre><code>## Calculating cluster 1</code></pre>
<pre><code>## Calculating cluster 2</code></pre>
<pre><code>## Calculating cluster 3</code></pre>
<pre><code>## Calculating cluster 4</code></pre>
<pre><code>## Calculating cluster 5</code></pre>
<pre><code>## Calculating cluster 6</code></pre>
<pre><code>## Calculating cluster 7</code></pre>
<pre><code>## Calculating cluster 8</code></pre>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="seurat.html#cb140-1" tabindex="-1"></a>pbmc.markers <span class="sc">%&gt;%</span></span>
<span id="cb140-2"><a href="seurat.html#cb140-2" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb140-3"><a href="seurat.html#cb140-3" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## # A tibble: 7,019 × 7
## # Groups:   cluster [9]
##        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene     
##        &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;    
##  1 3.75e-112       1.21 0.912 0.592 5.14e-108 0       LDHB     
##  2 9.57e- 88       2.40 0.447 0.108 1.31e- 83 0       CCR7     
##  3 1.15e- 76       1.06 0.845 0.406 1.58e- 72 0       CD3D     
##  4 1.12e- 54       1.04 0.731 0.4   1.54e- 50 0       CD3E     
##  5 1.35e- 51       2.14 0.342 0.103 1.86e- 47 0       LEF1     
##  6 1.94e- 47       1.20 0.629 0.359 2.66e- 43 0       NOSIP    
##  7 2.81e- 44       1.53 0.443 0.185 3.85e- 40 0       PIK3IP1  
##  8 6.27e- 43       1.99 0.33  0.112 8.60e- 39 0       PRKCQ-AS1
##  9 1.16e- 40       2.70 0.2   0.04  1.59e- 36 0       FHIT     
## 10 1.34e- 34       1.96 0.268 0.087 1.84e- 30 0       MAL      
## # ℹ 7,009 more rows</code></pre>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="seurat.html#cb142-1" tabindex="-1"></a>cluster0.markers <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(rUMAP_pbmc, <span class="at">ident.1 =</span> <span class="dv">0</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.25</span>, <span class="at">test.use =</span> <span class="st">&quot;roc&quot;</span>, <span class="at">only.pos =</span> <span class="cn">TRUE</span>)</span>
<span id="cb142-2"><a href="seurat.html#cb142-2" tabindex="-1"></a></span>
<span id="cb142-3"><a href="seurat.html#cb142-3" tabindex="-1"></a><span class="fu">VlnPlot</span>(rUMAP_pbmc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;MS4A1&quot;</span>, <span class="st">&quot;CD79A&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="seurat.html#cb143-1" tabindex="-1"></a><span class="co"># you can plot raw counts as well</span></span>
<span id="cb143-2"><a href="seurat.html#cb143-2" tabindex="-1"></a><span class="fu">VlnPlot</span>(rUMAP_pbmc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;NKG7&quot;</span>, <span class="st">&quot;PF4&quot;</span>), <span class="at">slot =</span> <span class="st">&quot;counts&quot;</span>, <span class="at">log =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Warning: The `slot` argument of `VlnPlot()` is
## deprecated as of Seurat 5.0.0.
## ℹ Please use the `layer` argument
##   instead.
## This warning is displayed once every 8
## hours.
## Call
## `lifecycle::last_lifecycle_warnings()`
## to see where this warning was generated.</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-25-2.png" width="672" /></p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="seurat.html#cb145-1" tabindex="-1"></a><span class="fu">FeaturePlot</span>(rUMAP_pbmc, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;MS4A1&quot;</span>, <span class="st">&quot;GNLY&quot;</span>, <span class="st">&quot;CD3E&quot;</span>, <span class="st">&quot;CD14&quot;</span>, <span class="st">&quot;FCER1A&quot;</span>, <span class="st">&quot;FCGR3A&quot;</span>, <span class="st">&quot;LYZ&quot;</span>, <span class="st">&quot;PPBP&quot;</span>,</span>
<span id="cb145-2"><a href="seurat.html#cb145-2" tabindex="-1"></a>    <span class="st">&quot;CD8A&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-25-3.png" width="672" /></p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="seurat.html#cb146-1" tabindex="-1"></a>pbmc.markers <span class="sc">%&gt;%</span></span>
<span id="cb146-2"><a href="seurat.html#cb146-2" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb146-3"><a href="seurat.html#cb146-3" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb146-4"><a href="seurat.html#cb146-4" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb146-5"><a href="seurat.html#cb146-5" tabindex="-1"></a>    <span class="fu">ungroup</span>() <span class="ot">-&gt;</span> top10</span>
<span id="cb146-6"><a href="seurat.html#cb146-6" tabindex="-1"></a><span class="fu">DoHeatmap</span>(rUMAP_pbmc, <span class="at">features =</span> top10<span class="sc">$</span>gene) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="seurat.html#cb147-1" tabindex="-1"></a>new.cluster.ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Naive CD4 T&quot;</span>, <span class="st">&quot;CD14+ Mono&quot;</span>, <span class="st">&quot;Memory CD4 T&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="st">&quot;CD8 T&quot;</span>, <span class="st">&quot;FCGR3A+ Mono&quot;</span>,</span>
<span id="cb147-2"><a href="seurat.html#cb147-2" tabindex="-1"></a>    <span class="st">&quot;NK&quot;</span>, <span class="st">&quot;DC&quot;</span>, <span class="st">&quot;Platelet&quot;</span>)</span>
<span id="cb147-3"><a href="seurat.html#cb147-3" tabindex="-1"></a><span class="fu">names</span>(new.cluster.ids) <span class="ot">&lt;-</span> <span class="fu">levels</span>(rUMAP_pbmc)</span>
<span id="cb147-4"><a href="seurat.html#cb147-4" tabindex="-1"></a>top10_genes_pbmc <span class="ot">&lt;-</span> <span class="fu">RenameIdents</span>(rUMAP_pbmc, new.cluster.ids)</span>
<span id="cb147-5"><a href="seurat.html#cb147-5" tabindex="-1"></a><span class="fu">DimPlot</span>(top10_genes_pbmc, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>, <span class="at">pt.size =</span> <span class="fl">0.5</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="curriculum-vitae-ffr.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
