<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Portfolio FFR</title>
  <meta name="description" content="Portfolio FFR" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Portfolio FFR" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Portfolio FFR" />
  
  
  

<meta name="author" content="FFR" />


<meta name="date" content="2024-12-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="curriculum-vitae-ffr.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="single-cell-analysis-of-crohns-dissease.html"><a href="single-cell-analysis-of-crohns-dissease.html"><i class="fa fa-check"></i><b>1</b> Single cell analysis of Crohn’s dissease</a>
<ul>
<li class="chapter" data-level="1.1" data-path="single-cell-analysis-of-crohns-dissease.html"><a href="single-cell-analysis-of-crohns-dissease.html#libraries-used"><i class="fa fa-check"></i><b>1.1</b> Libraries used</a></li>
<li class="chapter" data-level="1.2" data-path="single-cell-analysis-of-crohns-dissease.html"><a href="single-cell-analysis-of-crohns-dissease.html#loading-in-the-data-set-from-the-cz-cellxgene-database"><i class="fa fa-check"></i><b>1.2</b> Loading in the data set from the CZ CELLxGENE database</a></li>
<li class="chapter" data-level="1.3" data-path="single-cell-analysis-of-crohns-dissease.html"><a href="single-cell-analysis-of-crohns-dissease.html#qc-subsetting"><i class="fa fa-check"></i><b>1.3</b> QC subsetting</a></li>
<li class="chapter" data-level="1.4" data-path="single-cell-analysis-of-crohns-dissease.html"><a href="single-cell-analysis-of-crohns-dissease.html#preperations"><i class="fa fa-check"></i><b>1.4</b> Preperations</a></li>
<li class="chapter" data-level="1.5" data-path="single-cell-analysis-of-crohns-dissease.html"><a href="single-cell-analysis-of-crohns-dissease.html#variable-features-plot"><i class="fa fa-check"></i><b>1.5</b> Variable features plot</a></li>
<li class="chapter" data-level="1.6" data-path="single-cell-analysis-of-crohns-dissease.html"><a href="single-cell-analysis-of-crohns-dissease.html#pca"><i class="fa fa-check"></i><b>1.6</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="curriculum-vitae-ffr.html"><a href="curriculum-vitae-ffr.html"><i class="fa fa-check"></i><b>2</b> Curriculum vitae FFR</a>
<ul>
<li class="chapter" data-level="2.1" data-path="curriculum-vitae-ffr.html"><a href="curriculum-vitae-ffr.html#personalia-verander-zodat-er-meer-is-over-bioinfo"><i class="fa fa-check"></i><b>2.1</b> Personalia verander zodat er meer is over bioinfo</a></li>
<li class="chapter" data-level="2.2" data-path="curriculum-vitae-ffr.html"><a href="curriculum-vitae-ffr.html#opleiding"><i class="fa fa-check"></i><b>2.2</b> Opleiding</a></li>
<li class="chapter" data-level="2.3" data-path="curriculum-vitae-ffr.html"><a href="curriculum-vitae-ffr.html#werkervaring"><i class="fa fa-check"></i><b>2.3</b> Werkervaring</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="seurat.html"><a href="seurat.html"><i class="fa fa-check"></i><b>3</b> Seurat</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Portfolio FFR</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Portfolio FFR</h1>
<p class="author"><em>FFR</em></p>
<p class="date"><em>2024-12-01</em></p>
</div>
<div id="single-cell-analysis-of-crohns-dissease" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">1</span> Single cell analysis of Crohn’s dissease<a href="single-cell-analysis-of-crohns-dissease.html#single-cell-analysis-of-crohns-dissease" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Crohn’s dissease is characterized by chronic inflammation of the gastrointestinal [GI] tract. This causes the patient to suffer from diarrhea, stomach aches, fevers, weight loss, tiredness and blood in the feaces.</p>
<p>This document got the data from CZ CELLxGENE database [<a href="https://cellxgene.cziscience.com/collections/2b02dff7-e427-4cdc-96fb-c0f354c099aa" class="uri">https://cellxgene.cziscience.com/collections/2b02dff7-e427-4cdc-96fb-c0f354c099aa</a>] and this data was collected in this [<a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC7060942/" class="uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC7060942/</a>] study.</p>
<p>The method of this study is to firstly isolate lamina propria cells from both uninflamed and inflamed biopsies of 11 patients. The the 22 ileal specimens where used for single-cell transcriptomics and a UMI identifier is used to differentiate the samples.</p>
<div id="libraries-used" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Libraries used<a href="single-cell-analysis-of-crohns-dissease.html#libraries-used" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-4"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-4" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-5"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-5" tabindex="-1"></a><span class="co">#install.packages(&quot;Seurat&quot;)</span></span>
<span id="cb1-6"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-6" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-7"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-7" tabindex="-1"></a><span class="co">#install.packages(&quot;patchwork&quot;)</span></span>
<span id="cb1-8"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-8" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-9"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-9" tabindex="-1"></a><span class="co">#install.packages(biomaRt)</span></span>
<span id="cb1-10"><a href="single-cell-analysis-of-crohns-dissease.html#cb1-10" tabindex="-1"></a><span class="co">#library(biomaRt)</span></span></code></pre></div>
</div>
<div id="loading-in-the-data-set-from-the-cz-cellxgene-database" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Loading in the data set from the CZ CELLxGENE database<a href="single-cell-analysis-of-crohns-dissease.html#loading-in-the-data-set-from-the-cz-cellxgene-database" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>I do have a local copy due to not necessarily wanting to fully rely on the database. The decision was eventually made to make the crohns_dissease_seurat object from the url provided in the database. Due to being unable to provide the data set in the same github repository as the other files in this portfolio. This is the best solution i could come up with wanting to make this open and reproducible science.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb2-1" tabindex="-1"></a>crohns_dissease_seurat <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">url</span>(<span class="st">&quot;https://datasets.cellxgene.cziscience.com/7afb64af-cd3b-44ed-bd2a-3d7bd8a98e13.rds&quot;</span>))</span></code></pre></div>
</div>
<div id="qc-subsetting" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> QC subsetting<a href="single-cell-analysis-of-crohns-dissease.html#qc-subsetting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Quality control is always a very important part of data analysis. As the saying goes “garbage in, garbage out”. The three metrics are number of features which is the number of unique genes detected low amount of features indicates low quality cells or droplets and at the other side of the spectrum high feature counts indicate cells clumping together. The number of counts is the second of the three, this variable indicates the number of total molecules in a cell. The third metric is percentage of mitochondrial reads. A high concentration of mitochondrial reads often indicates dying cells. The quality control is done by visualizing and sub-setting this is done in figure 1.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb3-1" tabindex="-1"></a><span class="co"># Visualize QC metrics as a violin plot</span></span>
<span id="cb3-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb3-2" tabindex="-1"></a><span class="fu">VlnPlot</span>(crohns_dissease_seurat, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeaturess_RNA&quot;</span>, <span class="st">&quot;nCounts_RNA&quot;</span>, <span class="st">&quot;percent_mito&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">pt.size =</span> <span class="cn">FALSE</span>)<span class="sc">+</span></span>
<span id="cb3-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb3-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">caption =</span> <span class="st">&quot;Figure 1. QC metrics vizualizing all cells and artifacts in the dataset.&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/visualizing%20the%20QC%20metrics-1.png" width="672" /></p>
<p>Filtering cells with 200–2500 features (genes) ensures that cells with sufficient gene detection are included and excluding low-quality or dead cells with few features. At the other side of the spectrum also stopping potential clumps of cells from being in the data set these clumps often exhibit a high number of features. The threshold of &lt;10% mitochondrial RNA likely removes cells in cellular stress or apoptosis. The mean amount of mitochondrial DNA varies between cell types.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb4-1" tabindex="-1"></a><span class="co">#setting the limits on exceptable values to select alowing some [10%] mitochondrial dna into the data set </span></span>
<span id="cb4-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb4-2" tabindex="-1"></a>subset_crohns_dissease <span class="ot">&lt;-</span> <span class="fu">subset</span>(crohns_dissease_seurat, <span class="at">subset =</span> nFeaturess_RNA <span class="sc">&gt;</span> <span class="dv">200</span> <span class="sc">&amp;</span> nFeaturess_RNA <span class="sc">&lt;</span> <span class="dv">2500</span> <span class="sc">&amp;</span> nCounts_RNA <span class="sc">&gt;</span> <span class="dv">200</span>  <span class="sc">&amp;</span>  percent_mito <span class="sc">&lt;</span> <span class="dv">10</span>)</span>
<span id="cb4-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="single-cell-analysis-of-crohns-dissease.html#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="single-cell-analysis-of-crohns-dissease.html#cb4-5" tabindex="-1"></a><span class="fu">VlnPlot</span>(subset_crohns_dissease, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">&quot;nFeaturess_RNA&quot;</span>, <span class="st">&quot;nCounts_RNA&quot;</span>, <span class="st">&quot;percent_mito&quot;</span>), <span class="at">ncol =</span> <span class="dv">3</span>, <span class="at">pt.size =</span> <span class="cn">FALSE</span>)<span class="sc">+</span></span>
<span id="cb4-6"><a href="single-cell-analysis-of-crohns-dissease.html#cb4-6" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">caption =</span> <span class="st">&quot;Figure 2. QC metrics vizualizing all cells and artifacts in the subset of the data set&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/subsetting%20the%20data-1.png" width="672" /></p>
<p>The data quality is good overall, with most cells falling within expected ranges. Filtering cells with low features, low total counts, or high mitochondrial content will further enhance the dataset quality and make it suitable for downstream analysis.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb5-1" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(subset_crohns_dissease, <span class="at">feature1 =</span> <span class="st">&quot;nCounts_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;percent_mito&quot;</span>)</span>
<span id="cb5-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb5-2" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(subset_crohns_dissease, <span class="at">feature1 =</span> <span class="st">&quot;nCounts_RNA&quot;</span>, <span class="at">feature2 =</span> <span class="st">&quot;nFeaturess_RNA&quot;</span>)</span>
<span id="cb5-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb5-3" tabindex="-1"></a>plot1 <span class="sc">+</span> <span class="fu">NoLegend</span>() <span class="sc">+</span> plot2 <span class="sc">+</span> <span class="fu">NoLegend</span>()<span class="sc">+</span></span>
<span id="cb5-4"><a href="single-cell-analysis-of-crohns-dissease.html#cb5-4" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">caption =</span> <span class="st">&quot;Figure 3. Scatterplots showing QC metrics distributions across cells&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/feuture%20scatter-1.png" width="672" /></p>
</div>
<div id="preperations" class="section level2 hasAnchor" number="1.4">
<h2><span class="header-section-number">1.4</span> Preperations<a href="single-cell-analysis-of-crohns-dissease.html#preperations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Normalization, making data human readable and preparing the data for future functions.<br />
Find variable features
The FindVariableFeatures function in Seurat is used to identify genes that exhibit high variability across cells in a given dataset, which are likely to be biologically significant using the selection method vst.</p>
<p>From the seurat website:</p>
<p>“ <strong>vst: First, fits a line to the relationship of log(variance) and log(mean) using local polynomial regression (loess). Then standardizes the feature values using the observed mean and expected variance (given by the fitted line). Feature variance is then calculated on the standardized values after clipping to a maximum (see clip.max parameter).</strong> ”</p>
<p>After that extracting the variable features and storing the top 7 as an object in R.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb6-1" tabindex="-1"></a><span class="co"># Normalize the data</span></span>
<span id="cb6-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb6-2" tabindex="-1"></a>norm_subset_crohns_dissease <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(subset_crohns_dissease)</span>
<span id="cb6-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="single-cell-analysis-of-crohns-dissease.html#cb6-4" tabindex="-1"></a><span class="co"># Find variable features</span></span>
<span id="cb6-5"><a href="single-cell-analysis-of-crohns-dissease.html#cb6-5" tabindex="-1"></a>FVF_crohns_dissease <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(norm_subset_crohns_dissease, <span class="at">selection.method =</span> <span class="st">&quot;vst&quot;</span>)</span>
<span id="cb6-6"><a href="single-cell-analysis-of-crohns-dissease.html#cb6-6" tabindex="-1"></a></span>
<span id="cb6-7"><a href="single-cell-analysis-of-crohns-dissease.html#cb6-7" tabindex="-1"></a>variable_features <span class="ot">&lt;-</span> <span class="fu">VariableFeatures</span>(FVF_crohns_dissease)</span>
<span id="cb6-8"><a href="single-cell-analysis-of-crohns-dissease.html#cb6-8" tabindex="-1"></a></span>
<span id="cb6-9"><a href="single-cell-analysis-of-crohns-dissease.html#cb6-9" tabindex="-1"></a>top7_gene_names <span class="ot">&lt;-</span> <span class="fu">head</span>(variable_features, <span class="dv">7</span>)</span></code></pre></div>
</div>
<div id="variable-features-plot" class="section level2 hasAnchor" number="1.5">
<h2><span class="header-section-number">1.5</span> Variable features plot<a href="single-cell-analysis-of-crohns-dissease.html#variable-features-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Feature plot is used to visualize the most variable genes in the dataset, using Seurat’s VariableFeaturePlot function. The plot displays the variation in gene expression across cells the top 7 genes are displayed with their gene symbol.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-1" tabindex="-1"></a><span class="co"># Plot variable features</span></span>
<span id="cb7-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-2" tabindex="-1"></a>plot3 <span class="ot">&lt;-</span> <span class="fu">VariableFeaturePlot</span>(FVF_crohns_dissease)</span>
<span id="cb7-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-3" tabindex="-1"></a></span>
<span id="cb7-4"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-4" tabindex="-1"></a><span class="co"># Label points with gene names using the build in solution instead of to 10 gene names</span></span>
<span id="cb7-5"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-5" tabindex="-1"></a>plot4 <span class="ot">&lt;-</span> <span class="fu">LabelPoints</span>(</span>
<span id="cb7-6"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-6" tabindex="-1"></a>  <span class="at">plot =</span> plot3,</span>
<span id="cb7-7"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-7" tabindex="-1"></a>  <span class="at">points =</span> <span class="fu">head</span>(variable_features, <span class="dv">7</span>),</span>
<span id="cb7-8"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-8" tabindex="-1"></a>  <span class="at">labels =</span> <span class="fu">head</span>(FVF_crohns_dissease<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>meta.features<span class="sc">$</span>gene_symbols, <span class="dv">7</span>),</span>
<span id="cb7-9"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-9" tabindex="-1"></a>  <span class="at">repel =</span> <span class="cn">TRUE</span></span>
<span id="cb7-10"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-10" tabindex="-1"></a>)</span>
<span id="cb7-11"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-11" tabindex="-1"></a></span>
<span id="cb7-12"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-12" tabindex="-1"></a><span class="co"># Display the plot</span></span>
<span id="cb7-13"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-13" tabindex="-1"></a>plot4<span class="sc">+</span></span>
<span id="cb7-14"><a href="single-cell-analysis-of-crohns-dissease.html#cb7-14" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">caption =</span> <span class="st">&quot;Figure 4. Variable features plot with top 7 genes&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
</div>
<div id="pca" class="section level2 hasAnchor" number="1.6">
<h2><span class="header-section-number">1.6</span> PCA<a href="single-cell-analysis-of-crohns-dissease.html#pca" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Scale Data function scales data before dimensional reduction techniques like PCA. It ensures that highly-expressed genes don’t dominate. Otherwise the results can be skewed.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb8-1" tabindex="-1"></a>all.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(FVF_crohns_dissease)</span>
<span id="cb8-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb8-2" tabindex="-1"></a></span>
<span id="cb8-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb8-3" tabindex="-1"></a>scaled_data <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(FVF_crohns_dissease, <span class="at">features =</span> all.genes)</span>
<span id="cb8-4"><a href="single-cell-analysis-of-crohns-dissease.html#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="single-cell-analysis-of-crohns-dissease.html#cb8-5" tabindex="-1"></a>pca_crohns <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(scaled_data, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> scaled_data)) </span></code></pre></div>
<p>Choosing the right amount of dimensions to analyse and visualize is important. This is done by manually looking at the elbow plot in Figure 5.</p>
<p>I choose 15 PC’s because the Elbow plot shows that the first few principal components (PCs) have much higher variance (standard deviation), while later PCs explain less of the variance. By selecting 15 components, you keep most of the important information while reducing the amount of processing power needing to be used.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb9-1" tabindex="-1"></a><span class="fu">ElbowPlot</span>(pca_crohns)<span class="sc">+</span></span>
<span id="cb9-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb9-2" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">caption =</span> <span class="st">&quot;Figure 5. Elbow plot showing the standard deviation of each principal component (PC)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb10-1" tabindex="-1"></a><span class="fu">DimPlot</span>(pca_crohns, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-3-1.png" width="672" />
## Reductional assays</p>
<p>Seurat generates heatmaps to visualize the expression of the genes per PC, PC1 through PC15.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb11-1" tabindex="-1"></a><span class="fu">DimHeatmap</span>(pca_crohns, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<p>Firstly the FindNeighbors and FindClusters functions are applied to make/calculate the clusters. Here after the UMAP was made using the RunUMAP function to project the data into a dimplot readable frame so it can be displayed into Figure 7 using dimplot. DimPlot function was employed to visualize the UMAP projection, with cells colored by their respective clusters.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb12-1" tabindex="-1"></a>umap_data_crohns <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(pca_crohns, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>)</span>
<span id="cb12-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb12-2" tabindex="-1"></a>umap_data_crohns <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(umap_data_crohns, <span class="at">resolution =</span> <span class="fl">0.5</span>)</span></code></pre></div>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 29915
## Number of edges: 937472
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.9079
## Number of communities: 19
## Elapsed time: 5 seconds</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb14-1" tabindex="-1"></a>umap_fig_crohns <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(umap_data_crohns, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>)</span>
<span id="cb14-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb14-2" tabindex="-1"></a></span>
<span id="cb14-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb14-3" tabindex="-1"></a><span class="fu">DimPlot</span>(umap_fig_crohns, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/UMAP%20visualization-1.png" width="672" /></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-1" tabindex="-1"></a><span class="co"># run this code if you do not have this RDS file</span></span>
<span id="cb15-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-2" tabindex="-1"></a></span>
<span id="cb15-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-3" tabindex="-1"></a></span>
<span id="cb15-4"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-4" tabindex="-1"></a><span class="co"># find markers for every cluster compared to all remaining cells, report only the positive ones</span></span>
<span id="cb15-5"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-5" tabindex="-1"></a><span class="co">#This took me 2 hours and 21 minutes</span></span>
<span id="cb15-6"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-6" tabindex="-1"></a></span>
<span id="cb15-7"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-7" tabindex="-1"></a><span class="co"># if wanting to do this yourself run this code first to be more efficient with your time</span></span>
<span id="cb15-8"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-8" tabindex="-1"></a><span class="co">#library(devtools)</span></span>
<span id="cb15-9"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-9" tabindex="-1"></a><span class="co">#devtools::install_github(&#39;immunogenomics/presto&#39;)</span></span>
<span id="cb15-10"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-10" tabindex="-1"></a></span>
<span id="cb15-11"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-11" tabindex="-1"></a>crohns.markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(umap_fig_crohns, <span class="at">only.pos =</span> <span class="cn">TRUE</span>)</span>
<span id="cb15-12"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-12" tabindex="-1"></a></span>
<span id="cb15-13"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-13" tabindex="-1"></a><span class="co"># Save the results in an .rds file</span></span>
<span id="cb15-14"><a href="single-cell-analysis-of-crohns-dissease.html#cb15-14" tabindex="-1"></a><span class="fu">saveRDS</span>(crohns.markers, <span class="at">file =</span> <span class="st">&quot;~/school/DSFB2/dsfb2_workflows_portfolio/data/crohns.markers.rds&quot;</span>)</span></code></pre></div>
<p>Making an enormous heatmap that takes the top five genes per cluster into account.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-1" tabindex="-1"></a>crohns.markers <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/school/DSFB2/dsfb2_workflows_portfolio/data/crohns.markers.rds&quot;</span>)</span>
<span id="cb16-2"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-2" tabindex="-1"></a></span>
<span id="cb16-3"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-3" tabindex="-1"></a></span>
<span id="cb16-4"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-4" tabindex="-1"></a><span class="do">##################### because most of the &quot;top genes&quot; all have a p-adjusted value of what the computer stores as zero I decided to arrange the table to also include the highest avg. log2 fold change</span></span>
<span id="cb16-5"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-5" tabindex="-1"></a></span>
<span id="cb16-6"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-6" tabindex="-1"></a><span class="co">#crohns.markers %&gt;%</span></span>
<span id="cb16-7"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-7" tabindex="-1"></a><span class="co">#    group_by(cluster) %&gt;%</span></span>
<span id="cb16-8"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-8" tabindex="-1"></a><span class="co">#    dplyr::filter(avg_log2FC &gt; 1) %&gt;% </span></span>
<span id="cb16-9"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-9" tabindex="-1"></a><span class="co">#    distinct(gene) %&gt;%</span></span>
<span id="cb16-10"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-10" tabindex="-1"></a><span class="co">#    slice_head(n = 5) %&gt;%</span></span>
<span id="cb16-11"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-11" tabindex="-1"></a><span class="co">#    ungroup() -&gt; top5</span></span>
<span id="cb16-12"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-12" tabindex="-1"></a></span>
<span id="cb16-13"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-13" tabindex="-1"></a></span>
<span id="cb16-14"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-14" tabindex="-1"></a></span>
<span id="cb16-15"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-15" tabindex="-1"></a>top5_genes <span class="ot">&lt;-</span> crohns.markers <span class="sc">%&gt;%</span></span>
<span id="cb16-16"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-16" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb16-17"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-17" tabindex="-1"></a>    <span class="fu">arrange</span>(p_val_adj, <span class="fu">desc</span>(avg_log2FC)) <span class="sc">%&gt;%</span>  <span class="co"># Sort by lowest p_val_adj and then highest avg_log2FC</span></span>
<span id="cb16-18"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-18" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb16-19"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-19" tabindex="-1"></a>    <span class="fu">distinct</span>(gene) <span class="sc">%&gt;%</span></span>
<span id="cb16-20"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-20" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-21"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-21" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb16-22"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-22" tabindex="-1"></a></span>
<span id="cb16-23"><a href="single-cell-analysis-of-crohns-dissease.html#cb16-23" tabindex="-1"></a><span class="fu">DoHeatmap</span>(umap_fig_crohns, <span class="at">features =</span> top5_genes<span class="sc">$</span>gene) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/cluster%20markers-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="curriculum-vitae-ffr.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
},
"fig_caption": true,
"self_contained": false
});
});
</script>

</body>

</html>
